SCSS docblock options

joonty opened the pull request:
Hi Jeremy,

I decided that the best way to allow users to override options was to do it on a file level. I came up with the idea of doc block comments at the top of the SCSS file, which could override all the global options (i.e. output style, debug, line comments), plus give the option of an output path.

Example:

``` css
/**
 * @outputPath ../www/site.css
 * @debugInfo true
 * @outputStyle compressed
 */

body {
    /* etc... */
}
```

I've added a README markdown file which is shown on the project page, which has a bit more explanation of the possible options. I've also updated the version number to 0.4.

My master branch has the project configuration for Netbeans 6.9.1, and I also have a netbeans-7.0 branch. The code I've added is identical for both, but you might want to add a netbeans-7.0 branch to your repo too.

Try it out and let me know what you think!

Thanks

joonty committed:
Updated project XML for Netbeans 7

joonty committed:
Started doc block method

joonty committed:
Doc block parser working

joonty committed:
Normalized output file path

joonty committed:
Replaced obsolete Hashtable with HashMap

joonty committed:
Added README and updated version number

joonty committed:
Added images, updated README

joonty committed:
Updated images for README

jmarsault commented:
Hi Jon,

Thank you for your great job! It is a good idea to override the default
settings with the doc block comments.

I have some questions about it :

The tag "@outputPath" is not implemented in your code, is it a mistake
or you wanted to write "@outputFile"?
I have also a null pointer exception when the tag is not recognized. I have
not yet located the npe. It appears only when the variable DEBUGENABLED is
set to false.

java.lang.NullPointerException
at org.netbeans.core.output2.OutWriter.printANSI(OutWriter.java:521)
at org.netbeans.core.output2.OutWriter.print(OutWriter.java:483)
at org.netbeans.core.output2.ErrWriter.print(ErrWriter.java:211)
at org.netbeans.core.output2.ErrWriter.println(ErrWriter.java:165)
at org.netbeans.modules.scss.engine.Engine$1.run(Engine.java:69)
at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1424)
[catch] at
org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:1968)

It could be a good idea to ignore some tags like "@autor"? Maybe the
invalidDocBlocException is useless?

Thanks for your help,
Jeremy

Le 1 mars 2012 15:30, Jonathan Cairns <
reply@reply.github.com

> a crit :
> 
> Hi Jeremy,
> 
> I decided that the best way to allow users to override options was to do
> it on a file level. I came up with the idea of doc block comments at the
> top of the SCSS file, which could override all the global options (i.e.
> output style, debug, line comments), plus give the option of an output path.
> 
> Example:
> 
> ``` css
> /**
>  * @outputPath ../www/site.css
>  * @debugInfo true
>  * @outputStyle compressed
>  */
> 
> body {
>    /* etc... */
> }
> ```
> 
> I've added a README markdown file which is shown on the project page,
> which has a bit more explanation of the possible options. I've also updated
> the version number to 0.4.
> 
> My master branch has the project configuration for Netbeans 6.9.1, and I
> also have a netbeans-7.0 branch. The code I've added is identical for both,
> but you might want to add a netbeans-7.0 branch to your repo too.
> 
> Try it out and let me know what you think!
> 
> Thanks
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/joonty/scss-editor master
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/jmarsault/scss-editor/pull/1
> 
> -- Commit Summary --
> - Updated project XML for Netbeans 7
> - Started doc block method
> - Doc block parser working
> - Normalized output file path
> - Replaced obsolete Hashtable with HashMap
> - Added README and updated version number
> - Added images, updated README
> 
> -- File Changes --
> 
> A README.md (45)
> A images/options.png (0)
> A images/styles.png (0)
> M scss.editor/manifest.mf (2)
> M scss.editor/nbproject/project.properties (48)
> M scss.editor/nbproject/project.xml (16)
> A
> scss.editor/src/org/netbeans/modules/languages/scss/parser/ScssDocblockParser.java
> (252)
> M scss.editor/src/org/netbeans/modules/scss/engine/Engine.java (20)
> 
> -- Patch Links --
> 
>  https://github.com/jmarsault/scss-editor/pull/1.patch
>  https://github.com/jmarsault/scss-editor/pull/1.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jmarsault/scss-editor/pull/1

joonty commented:
Hi Jeremy,

You're absolutely right - I meant to put @outputFile in the README! I've updated that now.

Also, I didn't see the same issue with the npe, but it could be the Netbeans version. I've tried it on 7.0 and 7.1 successfully. Having said that, I was throwing an exception (which was then caught) if an unknown doc block was found. I removed this, as it was a bit unnecessary, so maybe that's fixed something else.

If it's still happening for you try setting `DEBUGENABLED = true` in the ScssDocblockParser class - that should give a bit more information about what's happening.

Thanks,
Jon

ssavajols commented:
Hi,

Where could we get a full list of ScssDocblock parameters ? 

Thx.
